src/pomf/controllers/upload_controller.cr

```diff
@@ -52,13 +52,29 @@ module Pomf
 
           size, hash = write_file(file_path, io)
 
-          url = URI.parse Pomf.upload_url
+          Pomf.db.connection do |db|
+            file_exists = db.exec("SELECT FROM uploads WHERE hash = $1", [hash]).rows
+
+            if file_exists.size > 0
+              file = Models::Upload.where("hash = $1", [hash])
```

Comment:
This will return nil if not found, so there's no need for the 2 SQL queries. Also call this `old_file` for clarity.

```cr
if old_file = Models::Upload.where("hash = $1", [hash])
  if File.exists?(...)
...
```